File: R/plot_l96_var.R
source("config_l96.R")

con <- file("l96_var.dat", "rb")
l2 <- readBin(con, "numeric", nc)
close(con)
title <- paste("4DVar", "step=", a)
plot(seq(1, nt, by=nw), l2, type="l", lwd=2,
     main=title, xlab="time", ylab="l2", ylim=c(0, 3))
abline(h=1, lt=2)
abline(h=mean(l2), lt=3)
legend("topleft", legend=c(
  "analysis", paste("mean", format(mean(l2), digits=3)), "observation"),
  lty=c(1, 3, 2))

File: Python/plot_l96_var.py
import numpy as np
import matplotlib.pyplot as plt
from config_l96 import nt, r, nw, a

plt.rcParams["font.size"] = 18

l2 = np.fromfile("l96_var.dat")

title = f"Var step={a}"
fig, ax = plt.subplots()
ax.plot(range(1, nt+1, nw), l2, color="k", label="analysis")
ax.axhline(y=np.sqrt(r), lw=2, ls='--', color="k", label="observation")
ax.axhline(y=np.mean(l2), lw=2, ls=':', color="k",
           label=f"mean {np.mean(l2):.3f}")
ax.set_title(title)
ax.set_xlabel("time")
ax.set_ylabel("L2")
ax.set_ylim(0, 3)
ax.legend(loc="upper left")
plt.show()


